### PR TITLE
GET /teams from RTS instead of config.json

### PR DESCRIPTION
**Pull request by lukebarnard1**
Now that the RTS contains config for teams, use GET /teams to get that information so that users will see be able to register as a team (but not yet auto-join rooms, be sent to welcome page or be tracked as a referral).

### CODE REVIEW

File: src/components/structures/login/Registration.js [hunk 1]
```diff
@@ -104,6 +96,27 @@ module.exports = React.createClass({
         this.registerLogic.setIdSid(this.props.idSid);
         this.registerLogic.setGuestAccessToken(this.props.guestAccessToken);
         this.registerLogic.recheckState();
+
+        if (this.props.teamServerConfig &&
+        this.props.teamServerConfig.teamServerURL &&
+        !this._rtsClient) {
```
Comment (dbkr) [inline src/components/structures/login/Registration.js]: Our usual indenting style here would be:

```
if (
    this.props.teamServerConfig &&
    [...]
) {
}
```

**Comment by lukebarnard1** [comment]
We could make the component `busy` until the data arrives?

**Comment by dbkr** [comment]
Probably ought to

File: src/components/structures/login/Registration.js [hunk 2]
```diff
@@ -115,6 +121,10 @@ module.exports = React.createClass({
                 });
             }, (err) => {
                 console.error('Error retrieving config for teams', err);
+            }).finally(() => {
+                this.setState({
+                    teamServerBusy: false,
+                });
```
Comment (dbkr) [inline src/components/structures/login/Registration.js]: You still want a .done() on the end to catch stray exceptions.

File: src/components/structures/login/Registration.js [hunk 3]
```diff
@@ -297,6 +307,8 @@ module.exports = React.createClass({
     },
 
     _getRegisterContentJsx: function() {
+        var Spinner = sdk.getComponent("elements.Spinner");
```
Comment (dbkr) [inline src/components/structures/login/Registration.js]: Also const please

File: src/components/structures/login/Registration.js (hunk 2 above)
Comment (lukebarnard1) [inline src/components/structures/login/Registration.js]: oh ok, thanks

**Pull request merged**